modules/apiserversource-kn.adoc

```diff
@@ -0,0 +1,143 @@
+// Module included in the following assemblies:
+//
+// serverless/knative_eventing/serverless-kn-source.adoc
+
+[id="apiserversource-kn_context"]
+= Using the ApiServerSource with the Knative CLI (`kn`)
+
+This guide describes the steps required to create, manage and delete an ApiServerSource using `kn` commands.
+
+.Prerequisites
+
+* You will need to have a Knative Serving and Eventing installation.
+* You will need to have created the `default` broker in the same namespace that the ApiServerSource will be installed in.
+* You will need to have the `kn` CLI installed.
+
+.Procedure
+
+. Create a service account, cluster role and cluster binding for the ApiServerSource.
++
+You can do this by creating a file named `authentication.yaml` and copying the following sample code into it:
++
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: events-sa
+  namespace: default <1>
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: event-watcher
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - get
+      - list
+      - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-ra-event-watcher
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-watcher
+subjects:
+  - kind: ServiceAccount
+    name: events-sa
+    namespace: default <1>
+----
++
+<1> Adjust this namespace to the namespace that you have selected for installing ApiServerSource.
++
+[NOTE]
+====
+If you want to re-use an existing service account with the appropriate permissions, you must modify the `authentication.yaml` for that service account.
+====
++
+Create the service account, cluster role and cluster binding by entering the following command:
++
+----
+$ oc apply --filename authentication.yaml
+----
+
+. Create an ApiServerSource event source by entering the following command:
++
+----
+$ kn source apiserver create testevents --sink broker:default --resource "event:v1" --service-account events-sa
+----
+
+. To check that the ApiServerSource is set up correctly, create a Knative service that dumps incoming messages to its log by entering the following command:
++
+----
+$ kn service create event-display --image quay.io/openshift-knative/knative-eventing-sources-event-display
+----
+
+. Create a trigger from the `default` broker to filter events to the service created in the previous step by entering the following `kn` command:
++
+----
+$ kn trigger create event-display-trigger --broker default --sink svc:event-display \ source=dev.knative.samples/testevents \
```

Comment:
I am not sure what the "source=dev.knative.samples/testevents" is supposed to do here?  (I assume this is some artifact from some other tutorial, as the ApiServerSource we create above doesn't define any "dev.knative.samples"?

Reply:
:man_shrugging:

Reply:
delete it maybe? I'm not sure what it is either, might have ended up in there by accident if nobody recognises it?

Reply:
True, the `source` declaration looks weird (an won't work that way). Also the backslashed don't make sense.

I think it look like:

```
kn trigger create event-display-trigger --sink svc:event-display
```

That correspond to the `Trigger` resource in the YAML section. You can ommit `--broker default` here as this is the, well, default ;-) (but you can equally add it for more expressiveness)

Reply:
Thanks, Roland!